tests/test_cli.py

```diff
@@ -747,6 +748,55 @@ def test_done_recurring(runner, todo_factory, todos):
     assert todo.rrule == rrule
 
 
+def test_undo(runner, todo_factory, todos):
+    completed_at = datetime.datetime.now(tz=pytz.UTC)
+    percentage = 100
+    todo = todo_factory(completed_at=completed_at, percent_complete=percentage)
+
+    result = runner.invoke(cli, ["undo", "1"])
+    assert not result.exception
+
+    todo = next(todos(status="ANY"))
+    assert todo.percent_complete == 0
+    assert todo.is_completed is False
+    assert not todo.rrule
+
+
+@pytest.mark.xfail(reason="Not implemented")
```

Comment:
I think that recursion is now implemented, right?

Reply:
No, it is not.

There are multiple PRs regarding recursion. The most relevant here is #288 but there are some more listed in #485.